--- swftools-0.4.3/pdf2swf/SWFOutputDev.cc.orig	2002-08-17 11:23:46.000000000 +0200
+++ swftools-0.4.3/pdf2swf/SWFOutputDev.cc	2004-03-27 10:49:41.749263352 +0100
@@ -859,7 +859,7 @@
 	}
     }
     if(filename) {
-	for(i=0; i<T1_Get_no_fonts(); i++)
+	for(i=0; i<T1_GetNoFonts(); i++)
 	{
 	    char*fontfilename = T1_GetFontFileName (i);
 	    if(strstr(fontfilename, filename))
@@ -869,7 +869,7 @@
 	    }
 	}
     } else {
-	for(i=0; i<T1_Get_no_fonts(); i++)
+	for(i=0; i<T1_GetNoFonts(); i++)
 	{
 	    char*fontname = T1_GetFontName (i);
 	    if(!fontname) {
--- swftools-0.4.3/pdf2swf/pdf2swf.cc.orig	2002-08-26 22:34:07.000000000 +0200
+++ swftools-0.4.3/pdf2swf/pdf2swf.cc	2004-03-27 11:00:13.212266384 +0100
@@ -380,6 +380,8 @@
 	exit(0);
     }
 
+/* t1lib comes configured in PLD, so we can remove this whole _insecure_ code */
+#if 0
     logf("<verbose> reading font files from %s/fonts\n", DATADIR);
     //TODO: use tempnam here. Check if environment already contains a
     //T1LIB_CONFIG.
@@ -433,14 +435,16 @@
     fprintf(fi, "TYPE1=%s:.\n", t1searchpath);
     fclose(fi);
     fclose(db);
+#endif
     /* initialize t1lib */
     T1_SetBitmapPad( 16);
     if ((T1_InitLib(NO_LOGFILE)==NULL)){
 	fprintf(stderr, "Initialization of t1lib failed\n");
 	exit(1);
     }
+#if 0
     unlink("/tmp/t1lib.config.tmp");
-
+#endif
     pdfswf_init(filename, password);
     pdfswf_setoutputfilename(outputname);
 
--- swftools-0.4.3/pdf2swf/Makefile.in.orig	2004-03-27 10:48:31.000000000 +0100
+++ swftools-0.4.3/pdf2swf/Makefile.in	2004-03-27 11:01:01.333950776 +0100
@@ -33,8 +33,6 @@
 	$(mkinstalldirs) $(bindir)
 	$(mkinstalldirs) $(man1dir)
 	@file=pdf2swf;$(INSTALL_BIN);$(INSTALL_MAN1)
-	$(mkinstalldirs) $(pkgdatadir)/fonts
-	@for file in $(fonts);do $(INSTALL_DATA) fonts/$$file $(pkgdatadir)/fonts/$$file;done
 	
 uninstall:
 	@file=pdf2swf;$(UNINSTALL_BIN);$(UNINSTALL_MAN1)
