--- swftools-0.9.0/lib/modules/swftext.c~	2009-01-27 18:18:37.000000000 +0100
+++ swftools-0.9.0/lib/modules/swftext.c	2010-01-18 14:29:10.530704314 +0100
@@ -447,8 +447,12 @@
 
 		if (id == fid) {
 		    if (jobs & FEDTJ_PRINT) {
-			int code = f->glyph2ascii[glyph];
-			printf("%lc", code);
+			if ( f->glyph2ascii ) {
+				int code = f->glyph2ascii[glyph];
+				printf("%lc", code);
+			} else {
+				printf("<?%d>", glyph);
+			}
 		    }
 		    if (jobs & FEDTJ_MODIFY)
 			f->glyph[glyph].advance = adv * 20;	//?
