--- swftools-0.4.3/src/Makefile.in.orig	2003-01-07 09:33:02.000000000 +0100
+++ swftools-0.4.3/src/Makefile.in	2004-03-27 10:48:40.241613936 +0100
@@ -25,19 +25,21 @@
 	$(C) jpeg2swf.c -o $@
 
 wav2swf: wav2swf.o wav.o
-	$(L) wav2swf.o wav.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) wav2swf.o wav.o -o $@ ../lib/librfxswf.la $(LIBS)
 png2swf: png2swf.o
-	$(L) png2swf.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) png2swf.o -o $@ ../lib/librfxswf.la $(LIBS)
 swfcombine: swfcombine.o
-	$(L) swfcombine.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) swfcombine.o -o $@ ../lib/librfxswf.la $(LIBS)
 swfdump: swfdump.o
-	$(L) swfdump.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) swfdump.o -o $@ ../lib/librfxswf.la $(LIBS)
 swfextract: swfextract.o
-	$(L) swfextract.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) swfextract.o -o $@ ../lib/librfxswf.la $(LIBS)
 swfstrings: swfstrings.o
-	$(L) swfstrings.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) swfstrings.o -o $@ ../lib/librfxswf.la $(LIBS)
 jpeg2swf: jpeg2swf.o
-	$(L) jpeg2swf.o -o $@ ../lib/librfxswf.a $(LIBS)
+	$(SL) jpeg2swf.o -o $@ ../lib/librfxswf.la $(LIBS)
+
+INSTALL_PROGRAM=$(LIBTOOL) --mode=install install -c
 
 install:
 	$(mkinstalldirs) $(bindir)
--- swftools-0.4.3/avi2swf/Makefile.in.orig	2003-01-08 18:55:51.000000000 +0100
+++ swftools-0.4.3/avi2swf/Makefile.in	2004-03-27 10:48:16.739186848 +0100
@@ -9,7 +9,9 @@
 	$(CC) `avifile-config --cflags` $< -o $@
 
 avi2swf: avi2swf.o
-	$(LL) avi2swf.o -o avi2swf ../lib/librfxswf.a $(LIBS) `avifile-config --libs`
+	$(SLL) avi2swf.o -o avi2swf ../lib/librfxswf.la $(LIBS) `avifile-config --libs`
+
+INSTALL_PROGRAM=$(LIBTOOL) --mode=install install -c
 
 install:
 	$(mkinstalldirs) $(bindir)
--- swftools-0.4.3/pdf2swf/Makefile.in.orig	2003-01-07 09:36:59.000000000 +0100
+++ swftools-0.4.3/pdf2swf/Makefile.in	2004-03-27 10:48:31.247981176 +0100
@@ -19,7 +19,7 @@
 	$(CC) -DDATADIR=\"$(pkgdatadir)\" -I../lib -Ittf2pt1 -Ixpdf swfoutput.cc -o $@
 
 pdf2swf: pdf2swf.o SWFOutputDev.o swfoutput.o encodings.o spline.o xpdf/pdflib.a ttf2pt1/ttf2pt1.a
-	$(LL) pdf2swf.o SWFOutputDev.o swfoutput.o encodings.o spline.o -o pdf2swf ../lib/librfxswf.a ttf2pt1/ttf2pt1.a xpdf/libpdf.a $(LIBS) 
+	$(SLL) pdf2swf.o SWFOutputDev.o swfoutput.o encodings.o spline.o -o pdf2swf ../lib/librfxswf.la ttf2pt1/ttf2pt1.a xpdf/libpdf.a $(LIBS) 
 
 xpdf/pdflib.a:
 	cd xpdf;$(MAKE);cd ..
@@ -27,6 +27,8 @@
 ttf2pt1/ttf2pt1.a:
 	cd ttf2pt1;$(MAKE);cd ..
 
+INSTALL_PROGRAM=$(LIBTOOL) --mode=install install -c
+
 install:
 	$(mkinstalldirs) $(bindir)
 	$(mkinstalldirs) $(man1dir)
--- swftools-0.4.3/lib/Makefile.in.orig	2003-01-07 18:20:33.000000000 +0100
+++ swftools-0.4.3/lib/Makefile.in	2004-03-27 10:46:37.341297624 +0100
@@ -3,26 +3,25 @@
 top_srcdir = @top_srcdir@
 include ../Makefile.common
 
-all: librfxswf.a
+all: librfxswf.la
 
-lame_objects =  lame/psymodel.o lame/fft.o lame/newmdct.o lame/quantize.o lame/takehiro.o lame/reservoir.o lame/quantize_pvt.o lame/vbrquantize.o lame/encoder.o lame/id3tag.o lame/version.o lame/tables.o lame/util.o lame/bitstream.o lame/set_get.o lame/VbrTag.o lame/lame.o
+lame_objects =  -lmp3lame
 rfxswf_modules =  modules/swfbits.c modules/swfaction.c modules/swfdump.c modules/swfcgi.c modules/swfbutton.c modules/swftext.c modules/swftools.c modules/swfsound.c modules/swfshape.c modules/swfobject.c
 
-bitio.o: bitio.c bitio.h
-	$(C) bitio.c -o $@
-log.o: log.c log.h
-	$(C) log.c -o $@
-rfxswf.o: rfxswf.c rfxswf.h bitio.h log.h $(rfxswf_modules)
-	$(C) rfxswf.c -o $@
-
-$(lame_objects):
-	cd lame;make all;cd ..
-
-librfxswf.a: rfxswf.o bitio.o log.o $(lame_objects)
-	$(AR) r librfxswf.a rfxswf.o bitio.o log.o $(lame_objects)
-	$(RANLIB) librfxswf.a
+bitio.lo: bitio.c bitio.h
+	$(SC) bitio.c -o $@
+log.lo: log.c log.h
+	$(SC) log.c -o $@
+rfxswf.lo: rfxswf.c rfxswf.h bitio.h log.h $(rfxswf_modules)
+	$(SC) rfxswf.c -o $@
+
+librfxswf.la: rfxswf.lo bitio.lo log.lo
+	$(SL) -o librfxswf.la rfxswf.lo bitio.lo log.lo $(lame_objects) -ljpeg -lz
 
 install:
+	install -d $(DESTDIR)$(libdir)
+	$(LIBTOOL) --mode=install install librfxswf.la $(DESTDIR)$(libdir)
+
 uninstall:
 
 clean: 
--- swftools-0.4.3/swfs/Makefile.in.orig	2003-01-07 09:31:58.000000000 +0100
+++ swftools-0.4.3/swfs/Makefile.in	2004-03-27 10:48:00.249693632 +0100
@@ -15,11 +15,11 @@
 	$(C) -I../lib PreLoaderTemplate.c -o $@
 
 simple_viewer: simple_viewer.o
-	$(L) simple_viewer.o -o $@ ../lib/librfxswf.a $(LIBS) 
+	$(SL) simple_viewer.o -o $@ ../lib/librfxswf.la $(LIBS)
 keyboard_viewer: keyboard_viewer.o
-	$(L) keyboard_viewer.o -o $@ ../lib/librfxswf.a $(LIBS) 
+	$(SL) keyboard_viewer.o -o $@ ../lib/librfxswf.la $(LIBS)
 PreLoaderTemplate: PreLoaderTemplate.o
-	$(L) PreLoaderTemplate.o -o $@ ../lib/librfxswf.a $(LIBS) 
+	$(SL) PreLoaderTemplate.o -o $@ ../lib/librfxswf.la $(LIBS)
 
 simple_viewer.swf: $(programs)
 	@echo Calling ./keybard_viewer to create keyboard_viewer.swf
